File: android/src/main/java/org/ligi/gobandroid_hd/ui/BaseProfileActivity.java
package org.ligi.gobandroid_hd.ui;

import android.os.Bundle;
import android.widget.EditText;
import butterknife.BindView;
import butterknife.ButterKnife;
import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.ui.application.GobandroidFragmentActivity;

public class BaseProfileActivity extends GobandroidFragmentActivity {

    @BindView(R.id.username_edit)
    EditText username_et;

    @BindView(R.id.rank_edit)
    EditText rank_et;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.profile);
        setTitle(R.string.profile);

        getSupportActionBar().setDisplayShowTitleEnabled(true);

        ButterKnife.bind(this);

        rank_et.setText(settings.getRank());
        username_et.setText(settings.getUsername());
    }


    @Override
    protected void onPause() {
        settings.setRank(rank_et.getText().toString());
        settings.setUsername(username_et.getText().toString());
        super.onPause();
    }

}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/BaseProfileActivity.kt
package org.ligi.gobandroid_hd.ui

import android.os.Bundle
import kotlinx.android.synthetic.withCloud.profile.*
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.ui.application.GobandroidFragmentActivity

class BaseProfileActivity : GobandroidFragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.profile)
        setTitle(R.string.profile)

        supportActionBar?.setDisplayShowTitleEnabled(true)

        rank_edit.setText(GoPrefs.rank)
        username_edit.setText(GoPrefs.username)
    }


    override fun onPause() {
        GoPrefs.rank = rank_edit.text.toString()
        GoPrefs.username = username_edit.text.toString()
        super.onPause()
    }

}
